### DISCUSSION TITLE
Support for clean AVR(completely without Arduino's libraries)?

### DISCUSSION THREAD

**Discussion by voltangle** [jgromes/RadioLib#235] [Q&A] [answered]
Hello!

I really like this library, it is pretty useful and great, but I can't use it in my AVR project, because it depends on Arduino's core, when I am using clean AVR. Are there plans for supporting clean AVR?

**Comment by jgromes** [comment, marked as answer]
I considered this in the past and it could be possible, the main issue is that to get this working, one would essentially have to create their own Arduino core implementation for the target MCU, since Arduino is used as a hardware abstraction layer here.

There's a way to make it a bit easier, by creating thin wrappers around every single function and type from Arduino core. This is already partially implemented, for example, most of the library doesn't use Arduino `pinMode()` function, and is using `Module::pinMode()` wrapper - the only place where Arduino `pinMode()` is called is from inside this wrapper method. So the idea was that the methods from `Module` class could contain custom implementations that would behave like their Arduino counterparts.

However, I never quite finished that, there's still a couple places that aren't well separated from Arduino core - e.g. `SX126x::SPItransfer()`, which directly uses `SPIClass` and `SPISettings` from Arduino core.
